### PR TITLE
fix: added self-contained styles for action bar

### LINKED ISSUE #211: Self-contained style for action-bar
Refactoring to make the action-bar component support self-contained style.

### PR DESCRIPTION
**Pull request by droshev**
fixes #211

### CODE REVIEW

**Comment by droshev** [comment]
Maybe we can create separate test files when we have combined components - e.g. action bar with popover as we may want to have those combinations in test handy for future testing
@jbadan @greg-a-smith @hertweckhr1 @InnaAtanasova what do you think?

File: test/templates/action-bar/index.njk [hunk 1]
```diff
@@ -12,87 +10,32 @@
     .fd-action-bar {
         margin: 0 20px;
     }
```
Comment (jbadan) [inline test/templates/action-bar/index.njk]: Can we use the placeholder buttons already in layout-shared: https://github.com/SAP/fundamental-styles/blob/master/test/resources/layout-shared.css#L6

Comment (droshev) [reply test/templates/action-bar/index.njk]: done!

File: components/action-bar.scss [hunk 2]
```diff
@@ -0,0 +1,6 @@
+$fd-icons-path: "../scss/icons/";
+$fd-fonts-path: "../scss/fonts/";
+
+@import "../scss/components/action-bar";
+@import "../scss/icons/icon";
```
Comment (greg-a-smith) [inline components/action-bar.scss]: This component shouldn't need icons.

File: test/resources/layout-shared.css [hunk 3]
```diff
@@ -10,3 +10,28 @@ body {
...
+    padding: 0;
+    border: none;
+    border-radius: 4px;
+    background-color: white;
+    min-height: 28px;
+    width: 26px;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+}
```
Comment (greg-a-smith) [inline test/resources/layout-shared.css]: Can we replace both `.test-btn-placeholder` and `.test-btn-back` with `.btn-playground`?

File: test/templates/action-bar/component.njk [hunk 4]
```diff
@@ -1,32 +1,27 @@
...
-    <div class="fd-action-bar__back">
-        <button class="fd-button--light sap-icon--nav-back"></button>
-    </div>
-    {%- endif %}
+{% set headlines = ['h2', 'h3', 'h4', 'h5', 'h6'] %}
+
+{%- macro action_bar(properties={}, aria={}, title_size="h3") %}
+<div class="fd-action-bar">
+{%- if properties.back_button %}
+    <div class="fd-action-bar__back"><div class="test-btn-back sap-icon--nav-back"></div></div>
```
Comment (greg-a-smith) [inline test/templates/action-bar/component.njk]: Can we replace `test-btn-back` with `btn-playground`?  Also, we should not be using the class `sap-icon--nav-back` in the playground examples.

File: test/templates/action-bar/index.njk [hunk 5]
```diff
@@ -1,98 +1,16 @@
...
-                { "label": "Option 2" },
-                { "label": "Option 3" },
-                { "label": "Option 4" }
-                ]
-            })
-        }, modifier={ block: ['left']})
-    }}
-{%- endset %}
+{%- macro render_txt_button() %}<div class="test-btn-placeholder">Button</div>{%- endmacro %}
+{%- macro render_icon_button() %}<div class="test-btn-placeholder sap-icon--overflow"></div>{%- endmacro %}
```
Comment (greg-a-smith) [inline test/templates/action-bar/index.njk]: Can we replace `test-btn-placeholder` with `btn-playground`?  Also, we shouldn't be using the class `sap-icon--overflow` in the playground.

**Comment by greg-a-smith** [comment]
I pushed changes for the last two items noted.

**Pull request merged**